include/fastdds/dds/xtypes/dynamic_types/DynamicTypeBuilderFactory.hpp

```diff
@@ -0,0 +1,209 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FASTDDS_DDS_XTYPES_DYNAMIC_TYPES_DYNAMIC_TYPE_BUILDER_FACTORY_HPP
+#define FASTDDS_DDS_XTYPES_DYNAMIC_TYPES_DYNAMIC_TYPE_BUILDER_FACTORY_HPP
+
+#include <memory>
+
+#include <fastdds/dds/core/ReturnCode.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/TypeDescriptor.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/Types.hpp>
+#include <fastrtps/fastrtps_dll.h>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+namespace xtypes {
+class TypeObject;
+} // namespace xtypes
+
+class RTPS_DllAPI DynamicTypeBuilderFactory : public std::enable_shared_from_this<DynamicTypeBuilderFactory>
+{
+public:
+
+    /*!
+     * Returns the singleton factory object
+     * @remark This method is non thread-safe.
+     * @return @ref DynamicTypeBuilderFactory reference.
+     */
+    static traits<DynamicTypeBuilderFactory>::ref_type get_instance();
+
+    /*!
+     * Resets the singleton reference.
+     * @return @ref ReturnCode_t
+     * @retval RETCODE_OK is always returned.
+     */
+    static ReturnCode_t delete_instance();
+
+    /*!
+     * Retrieves the cached @ref DynamicType reference associated to a given primitive
+     * @param[in] kind Type identifying the primitive type to retrieve.
+     * @return @ref DynamicType reference
+     */
+    virtual traits<DynamicType>::ref_type get_primitive_type(
+            TypeKind kind) = 0;
+
+    /*!
+     * Creates a new @ref DynamicTypeBuilder reference based on the given @ref TypeDescriptor state.
+     * @param[in] descriptor @ref TypeDescriptor to be copied.
+     * @return New @ref DynamicTypeBuilder reference.
+     */
+    virtual traits<DynamicTypeBuilder>::ref_type create_type(
+            traits<TypeDescriptor>::ref_type descriptor) = 0;
+
+    /*!
+     * Creates a new @ref DynamicTypeBuilder reference based on the given @ref DynamicType reference.
+     * @param[in] type @ref DynamicType reference to be used.
+     * @return New @ref DynamicTypeBuilder reference
+     */
+    virtual traits<DynamicTypeBuilder>::ref_type create_type_copy(
+            traits<DynamicType>::ref_type type) = 0;
+
+    /*!
+     * Creates a new @ref DynamicTypeBuilder reference based on the given @ref TypeObject instance.
+     * @remark Not implemented yet.
```

Comment:
The idea is to have this implemented and this was included in the planned work.

Reply:
Currently I'm not implementing it. Remove when it is implemented.

Reply:
Done in 8d81c49e0